optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/chained/KOptMove.java

```diff
@@ -172,26 +172,23 @@ public String getSimpleMoveTypeDescription() {
     }
 
     @Override
-    public boolean equals(Object o) {
+    public boolean equals(final Object o) {
         if (this == o) {
             return true;
-        } else if (o instanceof KOptMove) {
-            KOptMove<?> other = (KOptMove) o;
-            return new EqualsBuilder()
-                    .append(entity, other.entity)
-                    .append(values, other.values)
-                    .isEquals();
-        } else {
+        }
+        if (o == null || getClass() != o.getClass()) {
             return false;
         }
+        final KOptMove<?> kOptMove = (KOptMove<?>) o;
+        return Objects.equals(entity, kOptMove.entity) &&
+                Arrays.equals(values, kOptMove.values);
     }
 
     @Override
     public int hashCode() {
-        return new HashCodeBuilder()
-                .append(entity)
-                .append(values)
-                .toHashCode();
+        int result = Objects.hash(entity);
+        result = 31 * result + Arrays.hashCode(values);
```

Comment:
Any reason why this doesn't use Objects.hash()?

I am ok for changing the hashcode here as long as it desn't violate the equals contract.